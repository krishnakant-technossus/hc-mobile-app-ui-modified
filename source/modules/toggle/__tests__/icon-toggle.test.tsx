// npx jest ./source/modules/toggle/__tests__/icon-toggle.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {Text} from 'react-native';

import IconToggle from '../icon-toggle';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

let options;

describe('IconToggle', () => {
  beforeEach(() => {
    options = [
      {
        label: 'Laughing',
        onPress: jest.fn(),
        content: selected => {
          return selected ? (
            <Text>ðŸ˜‚</Text>
          ) : (
            <Text style={{opacity: 0.5}}>ðŸ˜‚</Text>
          );
        },
        value: 1,
      },
      {
        label: 'Crying',
        onPress: jest.fn(),
        content: selected => {
          return selected ? (
            <Text>ðŸ˜­</Text>
          ) : (
            <Text style={{opacity: 0.5}}>ðŸ˜­</Text>
          );
        },
        value: 2,
      },
    ];
  });

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <IconToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const iconToggle = wrapper.getByTestId('icon-toggle');

      expect(iconToggle.children).toHaveLength(options.length);

      expect(iconToggle).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the children correctly', () => {
      const {getByText, queryAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <IconToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const toggleButtons = queryAllByTestId('icon-toggle-button');

      expect(toggleButtons).toHaveLength(options.length);

      const option1 = getByText(options[0].label);
      const option2 = getByText(options[1].label);

      expect(option1).toBeDefined();
      expect(option2).toBeDefined();
    });
  });

  describe('Interactions', () => {
    it('should call onPress function on click of icon button', () => {
      const {getAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <IconToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const buttons = getAllByTestId('icon-toggle-button');

      const [button1] = buttons;

      fireEvent.press(button1);

      expect(options[0].onPress).toHaveBeenCalledTimes(1);
    });
  });
});
