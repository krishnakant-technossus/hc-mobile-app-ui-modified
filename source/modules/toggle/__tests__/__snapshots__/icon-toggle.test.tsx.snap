// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconToggle Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
  testID="icon-toggle"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon-toggle-button"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 64,
            "height": 64,
            "justifyContent": "center",
            "shadowColor": "rgba(0, 0, 0, .15)",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 12,
            "width": 64,
          },
        ]
      }
      testID="icon-toggle-content"
    >
      <Text>
        ðŸ˜‚
      </Text>
    </View>
    <Text
      isMuted={false}
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontSize": 14,
            "fontWeight": "normal",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="icon-toggle-label"
    >
      Laughing
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="icon-toggle-button"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 64,
            "height": 64,
            "justifyContent": "center",
            "shadowColor": "#FFFFFF",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 12,
            "width": 64,
          },
        ]
      }
      testID="icon-toggle-content"
    >
      <Text
        style={
          Object {
            "opacity": 0.5,
          }
        }
      >
        ðŸ˜­
      </Text>
    </View>
    <Text
      isMuted={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontSize": 14,
            "fontWeight": "normal",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="icon-toggle-label"
    >
      Crying
    </Text>
  </View>
</View>
`;
